### PR TITLE
Integrate vm common 0 1 19

### PR DESCRIPTION
**Pull request by sasurobert**

### CODE REVIEW

File: arwen/host/execution_test_vmoutputs.go [hunk 1]
```diff
@@ -402,7 +402,7 @@ func expectedVMOutput_SameCtx_BuiltinFunctions_2() *vmcommon.VMOutput {
 		nil,
 	)
 	account.Balance = big.NewInt(1000)
-	account.BalanceDelta = big.NewInt(0).Sub(big.NewInt(1), big.NewInt(1))
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply arwen/host/execution_test_vmoutputs.go:405]: yes.

File: arwen/host/execution.go [hunk 2]
```diff
@@ -430,17 +429,20 @@ func (host *vmHost) callSCMethodIndirect() error {
...
 		return err
 	}
 
-	scAccount, _ := output.GetOutputAccount(runtime.GetSCAddress())
-	scAccount.BalanceDelta.Add(scAccount.BalanceDelta, valueToSend)
-	metering.UseGas(gasConsumed)
+	gasConsumed := input.GasProvided - vmOutput.GasRemaining
+	if vmOutput.GasRemaining < input.GasProvided {
+		metering.UseGas(gasConsumed)
+	}
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply arwen/host/execution.go:443]: it is protected ok right now.

File: arwen/contexts/output.go [hunk 3]
```diff
@@ -286,6 +285,25 @@ func (context *outputContext) resolveReturnCodeFromError(err error) vmcommon.Ret
...
 
+func (context *outputContext) AddToActiveState(rightOutput *vmcommon.VMOutput) {
+	rightOutput.GasRemaining = 0
+	if rightOutput.GasRefund != nil {
+		rightOutput.GasRefund.Add(rightOutput.GasRefund, context.outputState.GasRefund)
+	}
+
+	for address, rightAccount := range rightOutput.OutputAccounts {
+		leftAccount, ok := context.outputState.OutputAccounts[address]
+		if !ok || rightAccount.BalanceDelta == nil || leftAccount.BalanceDelta == nil {
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply arwen/contexts/output.go:296]: deleted leftAccount == nil. as it should not happen.

File: arwen/host/execution.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (camilbancioiu) [reply arwen/host/execution.go:443]: Okay, but then builtin functions are not allowed to give gas to the SC. It's safe as long as that stays true. My suggestion was not correct for this case, anyway.

**Pull request merged**